# Fundamentals of Data Analysis_Assessment 2019

### Title and explaination
- Ref 
- 1. https://mc.ai/studying-suicide-rates-from-1985-to-2016-prediction-using-machine-learning/
- 2. https://www.geeksforgeeks.org
- 3. https://pbpython.com/pandas_dtypes.html (Clean data)
- 

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
df=pd.read_csv("suicide_rates_overview_1985_to_2016.csv")
#!pwd_needed this to confirm my file path to load the dataset

In [7]:
df.head()

country  year     sex          age  suicides_no  population  \
0  Albania  1987    male  15-24 years           21      312900   
1  Albania  1987    male  35-54 years           16      308000   
2  Albania  1987  female  15-24 years           14      289700   
3  Albania  1987    male    75+ years            1       21800   
4  Albania  1987    male  25-34 years            9      274300   

   suicides/100k pop country-year  HDI for year  gdp_for_year ($)   \
0               6.71  Albania1987           NaN      2,156,624,900   
1               5.19  Albania1987           NaN      2,156,624,900   
2               4.83  Albania1987           NaN      2,156,624,900   
3               4.59  Albania1987           NaN      2,156,624,900   
4               3.28  Albania1987           NaN      2,156,624,900   

   gdp_per_capita ($)       generation  
0                 796     Generation X  
1                 796           Silent  
2                 796     Generation X  
3                 796  G.I. Generation  
4                 796          Boomers

In [15]:
print (df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27820 entries, 0 to 27819
Data columns (total 12 columns):
country               27820 non-null object
year                  27820 non-null int64
sex                   27820 non-null object
age                   27820 non-null object
suicides_no           27820 non-null int64
population            27820 non-null int64
suicides/100k pop     27820 non-null float64
country-year          27820 non-null object
HDI for year          8364 non-null float64
 gdp_for_year ($)     27820 non-null object
gdp_per_capita ($)    27820 non-null int64
generation            27820 non-null object
dtypes: float64(2), int64(4), object(6)
memory usage: 2.5+ MB
None


In [19]:
print (("The following are the uniques values for Generation"), df.generation.unique())

The following are the uniques values for Generation ['Generation X' 'Silent' 'G.I. Generation' 'Boomers' 'Millenials'
 'Generation Z']


In [23]:
print (("The following are the unique values for the years:"), df.year.unique())
df['year'].dtype

The following are the unique values for the years: [1987 1988 1989 1992 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002
 2003 2004 2005 2006 2007 2008 2009 2010 1985 1986 1990 1991 2012 2013
 2014 2015 2011 2016]


dtype('int64')

In [3]:
df.describe()

year   suicides_no    population  suicides/100k pop  \
count  27820.000000  27820.000000  2.782000e+04       27820.000000   
mean    2001.258375    242.574407  1.844794e+06          12.816097   
std        8.469055    902.047917  3.911779e+06          18.961511   
min     1985.000000      0.000000  2.780000e+02           0.000000   
25%     1995.000000      3.000000  9.749850e+04           0.920000   
50%     2002.000000     25.000000  4.301500e+05           5.990000   
75%     2008.000000    131.000000  1.486143e+06          16.620000   
max     2016.000000  22338.000000  4.380521e+07         224.970000   

       HDI for year  gdp_per_capita ($)  
count   8364.000000        27820.000000  
mean       0.776601        16866.464414  
std        0.093367        18887.576472  
min        0.483000          251.000000  
25%        0.713000         3447.000000  
50%        0.779000         9372.000000  
75%        0.855000        24874.000000  
max        0.944000       126352.000000

- Requirement to clean this data! Clearly there are not 27820 countries in the world. From looking at df.head(), it is clear that a number of of indecies are repeated to take the year into account. It is neccessary to count the number of unique entires in Col "Country". The number of countries is compounded by year and age. 

In [4]:
df['year'].nunique()

32

In [25]:
# Examine the number of suicides per country in total
spc = suicides_per_country = []

for a in df['country'].unique():
    suicides_per_country.append(sum(df[df['country'] == a]['suicides_no']))
spc

[1970,
 11,
 82219,
 1905,
 101,
 70111,
 50073,
 1656,
 93,
 463,
 177,
 59892,
 62761,
 348,
 318,
 226613,
 36388,
 42,
 107561,
 40895,
 53080,
 6792,
 18429,
 41418,
 412,
 43687,
 15297,
 0,
 20660,
 11683,
 7034,
 304,
 33677,
 329127,
 3224,
 291262,
 12368,
 38,
 8149,
 3426,
 73891,
 1108,
 12574,
 11294,
 132060,
 184,
 806902,
 101546,
 53,
 966,
 13090,
 12770,
 28039,
 1958,
 27,
 20,
 585,
 3894,
 111139,
 423,
 472,
 50833,
 14383,
 2013,
 16992,
 33,
 3483,
 4783,
 21330,
 139098,
 24061,
 9043,
 574,
 261730,
 72777,
 1209742,
 0,
 230,
 124,
 4,
 24179,
 98,
 10089,
 13437,
 10615,
 7321,
 100202,
 55641,
 2166,
 37795,
 26217,
 110643,
 4039,
 10131,
 8624,
 319950,
 622,
 136805,
 1034013,
 13138,
 34803]

### In this data frame, country
- year from 
- sex: Male or female
- age:
- suicides_no
- population
- suicides/100k pop
- country-year
- HDI for year: The Human Development Index (HDI) is a statistic composite index of life expectancy, education, and per capita income indicators, which are used to rank countries into four tiers of human development. A country scores a higher HDI when the lifespan is higher, the education level is higher, and the gross national income GNI (PPP) per capita is higher
- gdp_for_year (in dollars)
- gdp_per_capita (in dollars)
- generation: Is a steriotypy of different attributes, they include 'Generation X', 'Silent', 'G.I. Generation', 'Boomers', 'Millenials', 'Generation Z'.